collagraph/cgx/cgx.py

```diff
@@ -299,7 +298,12 @@ def convert_node_to_args(node, *, names=None):
             # i, a and b, so we don't want to wrap those names with `_lookup()`
             name_collector = NameCollector()
             for generator in for_tree.generators:
-                name_collector.generic_visit(generator.target)
+                # Apparently, a node visitor does _not_ visit the root node...
+                # So instead, just add the name directly if the root is an ast.Name node
+                if isinstance(generator.target, ast.Name):
+                    name_collector.names.add(generator.target.id)
```

Comment:
I would recommend not even creating the name collector in this case, and simplify

Reply:
Well, it is also contained within a for-loop over the generators, so it is not that simple. 
I could make a set and pass it by reference to the NameCollector, so that I can add the name directly to the set. Is a bit cleaner indeed.

Reply:
Right.... Your current code probably performs a little better though.